src/platform/monitoring/DowntimeNotification/util/helpers.js

```diff
@@ -60,10 +60,22 @@ export function createGlobalMaintenanceWindow({
  * @param {Array} maintenanceWindows The raw JSON data from the API
  * @returns {Map}
  */
+
 export function createServiceMap(maintenanceWindows = []) {
   const serviceMap = new Map();
 
-  for (const maintenanceWindow of maintenanceWindows) {
+  // Maintenance windows should be sorted in ascending order
+  // so that when a single externalService has multiple upcoming
+  // maintenance windows, we can easily grab the one with the
+  // earliest startTime and ignore any others that we encounter
+  const sortedMaintenanceWindows = maintenanceWindows.sort((a, b) => {
```

Comment:
Could this be moved to a separate function with its own unit tests?

Reply:
It could, my aim was to make the smallest number of changes possible to this file but if you think it's needed then I can do that

Reply:
My bias tends to be towards breaking functional, testable chunks out as much as possible - I like how it makes logic flow easy to read down. That said, not a hill I will die on since this is a fairly small piece. Your choice 🙂

Reply:
I'm going to leave it as-is as this isn't one of my apps and I think that it's a pretty small piece of logic. I agree with the sentiment completely though